system/libraries/Cache/Cache.php

```
@@ -100,28 +100,18 @@ class CI_Cache extends CI_Driver_Library {
 	 */
 	public function __construct($config = array())
 	{
-		$default_config = array(
-			'adapter',
-			'memcached'
-		);
-
-		foreach ($default_config as $key)
+		if (isset($config['adapter']) && in_array($config['adapter'], $this->valid_drivers))
```

This will silently ignore any invalid driver and keep the library working with the 'dummy' one.